src/main/java/net/fabricmc/loom/decompilers/cache/JarWalker.java

```diff
@@ -127,6 +128,40 @@ public static List<ClassEntry> findClasses(FileSystemUtil.Delegate fs) throws IO
 		}
 	}
 
+	/**
+	 * Check if the given class file denotes and inner class and find the corresponding outer class name.
+	 */
+	private static String findOuterClass(FileSystemUtil.Delegate fs, String classFile) throws IOException {
```

Comment:
This could be sped up by directly returning null if `$` is not in the path.

Reply:
that would break on any class with the attributes present but with the `$` character obfuscated away - though I can't think of any Minecraft versions where that is the case 🤔

Reply:
Isn't this run on the named jar though? So if the names are broken, then it's just a sign of broken mappings (or missing mappings, but like you said, it shouldn't happen with obf names).

Reply:
I suppose, and as opposed to having it the other way around (with `$` without attributes), this wouldn't lead to a crash, right? I'd just mean that class isn't cached.